### PR TITLE
Bugfix/um 238 judy allocation not found

### PR DESCRIPTION
**Pull request by mcfadden8**
The fix for this is in judyL2Array.h.  The other changes were a merge changes made by @robinson96

### CODE REVIEW

File: src/umpire/util/AllocationMap.cpp [hunk 1]
```diff
@@ -116,6 +116,9 @@ AllocationMap::findRecord(void* ptr) const
             << " with size " << alloc_record->m_size);
 
       }
+      else {
+         alloc_record = nullptr;
```
Comment (davidbeckingsale) [inline src/umpire/util/AllocationMap.cpp]: Should keep this.

Comment (mcfadden8) [reply src/umpire/util/AllocationMap.cpp]: 👍

**Comment by mcfadden8** [comment]
> Can you rebase this branch and remove the miscellaneous changes that are also included in #61

This should be ready.  I also checked in the test tool for replaying allocation map operations from the replay.csv file since I use that program to confirm the fix.  I'll add documentation for that program under a separate PR.

**Comment by robinson96** [comment]
hey - I think this will cause issues with my PR, and now it's merged, so nvm I guess. This will pull many of my commits into develop and then the last commit undoes them, which will cause fun issues when I merge this in.

**Pull request merged**